### PR TITLE
Added the Model classes and tests

### LINKED ISSUE #30: [FEAT] Get master up to date with model elements
**Is your feature request related to a problem? Please describe.**

master is currently behind on having the model classes that the submissionUpload and pragya branches have been using. 

**Describe the solution you'd like**

This PR aims to bring these changes up to master so that the other branches won't have conflicts when merging. It also provides the necessary unit tests for the model.

**Acceptance criteria**

Running the application doesn't result in any bugs and all the tests still pass. 

**Additional context**

Michael said to make this an issue as a way of documenting that this was something we needed to fix.

### PR DESCRIPTION
**Pull request by ryanribeiro**
## Proposed changes

> Describe the big picture of your changes. If it fixes a bug or resolves a feature request, be sure to link to that issue

This PR is to bring the master branch up to date on the underlying model classes that we use. It also has basic unit testing for the models that we are using. It doesn't break anything we previously used, however it is important to get everyone working from this version since it has the tests and the model classes.

## Types of changes

> What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [X] The title is described the contents of the PR
- [X] Unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (where applicable)
- [ ] I have moved issue to the “Need review” columns of projects

## Known issues

> Create issue for known issues and link them in the pull request

Closes #30

### CODE REVIEW

File: src/main/java/com/pproval/app/services/ServletInitializer.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
-package com.pproval.app;
+package com.pproval.app.services;
```
Comment (yeskunall) [inline src/main/java/com/pproval/app/services/ServletInitializer.java]: Wait, why was this moved inside `services`? :sweat_smile:

Comment (ryanribeiro) [reply src/main/java/com/pproval/app/services/ServletInitializer.java]: That's a good question actually... I definitely didn't make that change as part of what I just did, so it must have been something that was lingering from before.

File: src/main/java/com/pproval/app/model/Editor.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import javax.validation.constraints.NotNull;
+
+import lombok.*;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+public class Editor extends User {
```
Comment (yeskunall) [inline src/main/java/com/pproval/app/model/Editor.java]: This entity class is unneeded. As mentioned on Slack, we’re using in-memory auth built into [Spring Security](https://www.baeldung.com/java-config-spring-security). Please remove it.

File: src/main/java/com/pproval/app/model/Reviewer.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import javax.validation.constraints.NotNull;
+
+import lombok.*;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+public class Reviewer extends User {
```
Comment (yeskunall) [inline src/main/java/com/pproval/app/model/Reviewer.java]: This entity class is unneeded. Please remove it.

File: src/main/java/com/pproval/app/model/Submitter.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import javax.validation.constraints.NotNull;
+
+import lombok.*;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+public class Submitter extends User {
```
Comment (yeskunall) [inline src/main/java/com/pproval/app/model/Submitter.java]: This entity class is unneeded. Please remove it.

File: src/main/java/com/pproval/app/services/ServletInitializer.java (hunk 1 above)
Comment (yeskunall) [reply src/main/java/com/pproval/app/services/ServletInitializer.java]: Can you please undo this change? It adds no value.

File: src/main/java/com/pproval/app/model/Article.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  private Status status;
+
+  public enum Status {
+    SUBMITTED,
+    INREVIEW,
+    ACCEPTED,
+    REJECTED;
+  }
+
+  public Article(Submitter submitter, String name, String article){
```
Comment (yeskunall) [inline src/main/java/com/pproval/app/model/Article.java]:
```suggestion
  public Article(Submitter submitter, String name, String article) {
```

^ nit: format

File: src/main/java/com/pproval/app/model/Article.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
+package com.pproval.app.model;
```
Comment (yeskunall) [inline src/main/java/com/pproval/app/model/Article.java]: The `Article` entity needs to be changed up a little bit.

Because we no are no longer persisting users to a DB, we cannot use relational associations like `@OneToOne` and so on, because Hibernate won’t know which table to map to, because as you guessed it, there is no `User` table in Postgres.

So instead, what we can do instead is create a column `author` (or something like that) to map an article to a user.

File: src/main/java/com/pproval/app/services/ServletInitializer.java (hunk 1 above)
Comment (ryanribeiro) [reply src/main/java/com/pproval/app/services/ServletInitializer.java]: It's been fixed.

File: src/main/java/com/pproval/app/model/Editor.java (hunk 2 above)
Comment (ryanribeiro) [reply src/main/java/com/pproval/app/model/Editor.java]: File has been removed.

File: src/main/java/com/pproval/app/model/Submitter.java (hunk 4 above)
Comment (ryanribeiro) [reply src/main/java/com/pproval/app/model/Submitter.java]: File has been removed.

File: src/main/java/com/pproval/app/model/Reviewer.java (hunk 3 above)
Comment (ryanribeiro) [reply src/main/java/com/pproval/app/model/Reviewer.java]: File was removed.

File: src/main/java/com/pproval/app/model/Article.java (hunk 6 above)
Comment (ryanribeiro) [reply src/main/java/com/pproval/app/model/Article.java]: Changes have been made accordingly.

**Comment by ryanribeiro** [comment]
Project concluded, so I will close this PR.
